docs/code/getting-started.md

```diff
@@ -108,53 +108,69 @@ Apps on elementary OS are organized into standardized directories contained in y
 
 1. Create a new file in Scratch and save it as "gtk-hello.vala" inside your "src" folder
 
-2. First we must create the main function of our new GTK app. Type the following in your "gtk-hello.vala".
+2. First we create a special class called a `Gtk.Application`. `Gtk.Application` is a class that handles many important aspects of a Gtk+ app like app uniqueness and the application ID you need to identify your app to the notifications server. If you want some more details about `Gtk.Application`, [check out Valadoc](https://valadoc.org/gtk+-3.0/Gtk.Application). For now, type the following in your "gtk-hello.vala".
+
+        public class MyApp : Gtk.Application {
+
+            public MyApp () {
+                Object (
+                    application_id: "com.github.yourusername.yourrepositoryname",
+                    flags: ApplicationFlags.FLAGS_NONE
+                );
+            }
 
-        int main (string[] args) {
-            Gtk.init (ref args);
+            protected override void activate () {
+                var main_window = new Gtk.ApplicationWindow (this);
+                main_window.show_all ();
+            }
+
+            public static int main (string[] args) {
+                var app = new MyApp ();
+                return app.run (args);
+            }
         }
+        
+    Ready to test it out? Fire up your terminal and make sure you're in "~/Projects/gtk-hello/src". Then execute the following commands to compile and run your first Gtk+ app:
 
-3. Now, that we've initialized Gtk, we'll create a new window and declare a few properties of this window. You'll notice that most of these property names are pretty straightforward. Try to guess what each one does and we'll explain in a second. Type the following after the `Gtk.init` line, but before the last bracket:
+        ```bash
+        valac --pkg gtk+-3.0 gtk-hello.vala
+        ./gtk-hello
+        ```
 
-        var window = new Gtk.Window ();
-        window.title = "Hello World!";
-        window.set_border_width (12);
-        window.set_position (Gtk.WindowPosition.CENTER);
-        window.set_default_size (350, 70);
-        window.destroy.connect (Gtk.main_quit);
+    Did it work? If so, congratulations! If not, read over your source code again and look for errors. Also check the output of your terminal. Usually there is helpful output that will help you track down your mistake.
 
-        Gtk.main ();
-        return 0;
+3. Now, that we've created a new `Gtk.Application` with a `Gtk.ApplicationWindow` called `main_window`, let's declare a couple properties of this window. You'll notice that most of these property names are pretty straightforward. Try to guess what each one does and we'll explain in a second. Add the following properties to your `main_window`:
```

Comment:
Should specify exactly where to type these lines (within the activate function) rather than inferring it from the repeated line.